## KNN for Healthcare & Medical Diagnosis:

Steps:<br>
1. Import necessary libraries<br>
2. Load and explore the dataset<br>
3. Handle missing values (if any) & check data statistics<br>
4. Split data into training and testing sets<br>
5. Normalize features for better KNN performance<br>
6. Train KNN model with an initial K value<br>
7. Make predictions on test data<br>
8. Evaluate model performance using accuracy, confusion matrix, and classification report<br>
9. Optimize K value by testing different K values<br>
10. Retrain the model with the best K value for improved accuracy<br>

### 1. Import necessary libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# train test seperation 
from sklearn.model_selection import train_test_split
# KNN classificationa nd regresiion
from sklearn.neighbors import KNeighborsClassifier, KNeighborsRegressor

# Accuracy, confusion matrics check
from sklearn.metrics import accuracy_score, confusion_matrix

# normalization and stadardization
from sklearn.preprocessing import StandardScaler

### 2. Load and explore the dataset

In [3]:
df = pd.read_csv('heart.csv')
df.head()

age  sex  cp  trestbps  chol  fbs  ...  exang  oldpeak  slope  ca  thal  target
0   63    1   3       145   233    1  ...      0      2.3      0   0     1       1
1   37    1   2       130   250    0  ...      0      3.5      0   0     2       1
2   41    0   1       130   204    0  ...      0      1.4      2   0     2       1
3   56    1   1       120   236    0  ...      0      0.8      2   0     2       1
4   57    0   0       120   354    0  ...      1      0.6      2   0     2       1

[5 rows x 14 columns]

#### Dataset all information
This dataset includes the following features:<br>

age: The age of the patient.<br>
sex: Gender of the patient (0: female, 1: male).<br>
cp: Type of chest pain.<br>
trestbps: Resting blood pressure.<br>
chol: Serum cholesterol.<br>
fbs: Fasting blood sugar > 120 mg/dl.<br>
restecg: Resting electrocardiographic results.<br>
thalach: Maximum heart rate achieved.<br>
exang: Exercise induced angina.<br>
oldpeak: ST depression induced by exercise relative to rest<br>